File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* temp=head;
        int count{0}, index{0};
        while(temp!=NULL){
            temp=temp->next;
            count++;
        }
        
      index= (count/2)+1;
      temp=head;
      count=0;
      while(count!=index-1){
          temp=temp->next;
          count++;
      }
        
    return temp;
    }
};

File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        itr,numNodes=head,0
        while(itr):
            numNodes+=1
            itr=itr.next
            
        itr,i=head,0
        while(i<int(numNodes/2)):
            itr=itr.next
            i+=1
        return itr

        